File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int k, int score[], size_t score_len) {
    int* answer = (int*)malloc(sizeof(int)*(score_len));
    int rank_k[101];
    for(int i=0;i<k;i++){rank_k[i] = -1;}
    for(int i=0;i<score_len;i++){
        int idex = i<k ? i:k-1;
        int j = idex+1;
        for(;0<=j;j--){
            if(rank_k[j] < score[i])
            {
                rank_k[j+1] = rank_k[j];
            }
            else{
                break;
            }
        }
        rank_k[j+1] = score[i];
        answer[i] = rank_k[idex];
    }    
    
    return answer;
}

File: 프로그래머스/1/138477.명예의 전당 （1）-Python/명예의 전당 （1）.py
def solution(k, score):
    answer = []

    top_k = [-1]

    for j in range(len(score)):
        s = score[j]
        for i in range(len(top_k)):
            if top_k[i] < s:
                top_k = top_k[:i] + [s] + top_k[i:]
                answer.append(top_k[min(j, k-1)])
                break
    return answer